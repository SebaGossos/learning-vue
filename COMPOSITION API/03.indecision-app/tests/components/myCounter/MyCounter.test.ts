import { describe, test, expect } from 'vitest';
import MyCounter from '@/components/myCounter/MyCounter.vue'
import { mount } from '@vue/test-utils'

describe('<My Counter />', () => {
  test('Should match snapshot', () => {

    const wrapper = mount(MyCounter, {
      props: {
        value: 5,
      }
    });

    expect( wrapper.html() ).toMatchSnapshot()
  })
    
  test('renders the counter value correctly', () => {
    const value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value,
      }
    });
    const [ counterLabel, squareLabel ] = wrapper.findAll('h3')
    // expect(wrapper.find('[data-testid="square-label"]').text()).toContain(`Square: ${value * value}`)

    
    expect( counterLabel.text() ).toContain(`Counter: ${value}`)
    
    expect( squareLabel.text() ).toContain(`Square: ${value * value}`)
    
  });


  test('increments the counter when +1 button is clicked', async() => {

    const value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value,
      }
    });

    const btnIncrement = wrapper.find('button');
    await btnIncrement.trigger('click');

    const [ counterLabel, squareLabel ] = wrapper.findAll('h3')
    
    
    expect( counterLabel.text() ).toContain(`Counter: ${value + 1}`)
    expect( squareLabel.text() ).toContain(`Square: ${(value + 1) * (value + 1)}`)
    
  })

  test( 'Decrements the counter when -1 button is clicked twice', async () => {
    const value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value,
      }
    })

    const [ ,btnDecrement] = wrapper.findAll('button')

    for( let i=0; i < 2  ; i++) {
      await btnDecrement.trigger('click');
    }

    const [ counterLabel, squareLabel ] = wrapper.findAll('h3')

    expect( counterLabel.text() ).toContain(`Counter: ${value - 2}`)
    expect( squareLabel.text() ).toContain(`Square: ${(value - 2) * (value - 2)}`)
    
    
  })


}) 