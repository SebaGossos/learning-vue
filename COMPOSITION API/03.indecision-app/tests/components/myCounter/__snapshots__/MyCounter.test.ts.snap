// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<My Counter /> > Should match snapshot 1`] = `
"<section data-v-c2d32f30="">
  <h3 data-v-c2d32f30="">Counter: 5</h3>
  <h3 data-v-c2d32f30="" data-testid="square-label">Square: 25</h3>
  <div data-v-c2d32f30=""><button data-v-c2d32f30="" class="btn">+1</button><button data-v-c2d32f30="" class="btn">-1</button></div>
</section>"
`;
